addons/mail/static/src/discuss/call/common/call_settings.js

```diff
@@ -108,12 +106,12 @@ export class CallSettings extends Component {
         this.store.settings.useBlur = ev.target.checked;
     }
 
-    onChangeVideoFilterCheckbox(ev) {
+    onChangeShowOnlyVideo(ev) {
         const showOnlyVideo = ev.target.checked;
-        this.props.thread.showOnlyVideo = showOnlyVideo;
-        const activeRtcSession = this.props.thread.activeRtcSession;
+        this.store.settings.showOnlyVideo = showOnlyVideo;
+        const activeRtcSession = this.thread?.activeRtcSession;
         if (showOnlyVideo && activeRtcSession && !activeRtcSession.videoStream) {
-            this.props.thread.activeRtcSession = undefined;
+            this.thread.activeRtcSession = undefined;
         }
```

Comment:
Use of `thread` in this component is only used for these LOCs.

What about just closing all active RTC sessions that have no video stream? What I mean, all active RTC sessions could be available easily (e.g. `store.allActiveRtcSessions` as a declarative field, whether with `compute` or `onAdd`/`onDelete` side-effects, etc.), and then we only matter on the setting value and all active RTC sessions, not the thread at hand.

If we made this change, then we can remove everything about `this.thread`/`this.store.thread`/`this.props.thread` of this component, making it much simpler as it should.